modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -515,8 +521,18 @@ public void setFileIOFactory(FileIOFactory ioFactory) {
                                     name, oldVal, newVal));
                             }
 
-                            if (newVal != null && newVal)
+                            if (newVal != null && newVal) {
```

Comment:
A few linew below you check both values - `oldVal && !newVal`.
Here you check only `newVal`. 
Does it mean that disable record can be recorded twice? When oldVal == newVal

So, it seems you should keep logic of both check consistent.